README.md

```diff
@@ -53,9 +70,214 @@ This is the simplest case. Get the script in whatever way you prefer and include
+
+For iframe embeds, the `Promise` will be resolved with the `iframe` element immediately after it is injected into the DOM.
+
+#### Promise Rejection
+The `Promise` will be rejected due to any of the following conditions:
+
+- Bad parameters.
+- Any DOM manipulation error.
+- Failed script download.
+- Failed player creation.
+
+### Parameters
+The following are all the parameters that can be passed to the Brightcove Player Loader function. Required parameters are marked with an asterisk (\*).
+
+Parameters *must* be passed as an object.
+
+#### `accountId`\*
+* **REQUIRED**
+* *Type:* string | number
+* *Default:* `undefined`
+
+A Video Cloud account ID.
+
+#### `applicationId`
+* *Type:* string
+* *Default:* `undefined`
+
+The [application ID][bc-app-id] to be applied to the generated embed.
+
+#### `embedId`
+* *Type:* string
+* *Default:* `'default'`
+
+The Brightcove Player [embed ID][bc-embed-id] for the player. The default value is correct for most users.
+
+#### `embedOptions`
+* *Type:* Object
+* *Default:* `undefined`
+
+Used to provide certain options for embed generation. These include:
+
+##### `embedOptions.pip`
+* *Type:* boolean
+
+If `true`, will wrap the embed in a `<div class="vjs-pip-container">` element. This should be used when you need support for [the Brightcove Picture-in-Picture plugin][bc-pip].
+
+##### `embedOptions.playlist`
+* *Type:* boolean
+
+If `true`, will add a `<div class="vjs-playlist">` element after the embed. This should be used when you need support for [the Brightcove Playlist UI plugin][bc-playlists].
+
+##### `embedOptions.responsive`
+* *Type:* boolean | Object
+
+> **NOTE:** This approach should work for in-page embeds, but it is recommended that you use the Video.js `fluid` and/or `aspectRatio` options instead:
+>
+> ```js
+> options: {
+>   aspectRatio: '16:9'
+> }
+> ```
+
+Used customize the embed code to produce a responsively-sized player using [the "intrinsic ratio" wrapper approach][bc-responsive-iframe].
+
+When `true`, will produce a responsive embed code with a 16:9 aspect ratio that will fill its container.
+
+An object can be provided to customize this with the following sub-properties:
+
+- `aspectRatio`: A string Used to customize the aspect ratio to a value other than 16:9 (e.g., `'4:3'`).
+- `maxWidth`: A string used to restrain the maximum width of the player. This should use CSS units, such as pixels (e.g., `'960px'`).
+
+#### `embedType`
+* *Type:* string
+* *Default:* `'in-page'`
+
+The [embed code type][bc-embed-code-type] to produce. This parameter's value must be one of the following:
+
+* `'in-page'` or `brightcovePlayerLoader.EMBED_TYPE_IN_PAGE`: Also referred to as the advanced embed code, this injects the player directly into the top-level web page.
+* `'iframe'` or `brightcovePlayerLoader.EMBED_TYPE_IFRAME`: Also referred to as the basic embed code, this injects the player as an `<iframe>` element.
+
+#### `onEmbedCreated`
```

Comment:
Should there also be a function for a callback before the player is created but after the code is downloaded?

Reply:
Maybe? Honestly, I'd prefer to wait until there's a use-case for that. If you can think of one, let me know!

Reply:
Allows customers to add hooks, custom plugins, etc outside of the player. It's something that I needed to add in our chromecast player loader. Which may be able to switch to this at some point.

Reply:
in response to: https://github.com/brightcove/player-loader/pull/2/files#r203111887

Reply:
Ah, gotcha. That makes sense. Let's add that as a subsequent feature. Good idea! 👍